### PR TITLE
update links to get.helm.sh

### PR DESCRIPTION
**Pull request by bacongobbler**
related to #5663 

Signed-off-by: Matthew Fisher <matt.fisher@microsoft.com>

### CODE REVIEW

**Review by michelleN — APPROVED** [review]

**Review by hickeyma — CHANGES_REQUESTED** [review]
It looks good but for one small point. Should [release checklist](https://github.com/helm/helm/blob/dev-v3/docs/release_checklist.md) also be updated?

**Comment by bacongobbler** [comment]
Good point @hickeyma. I somehow missed that. I'll go and update those links as well. Thanks!

**Comment by bacongobbler** [comment]
In the process of updating links, I found an old section of the docs that wasn't updated after the fork. I removed that section. Thanks for the catch @hickeyma

**Review by hickeyma — CHANGES_REQUESTED** [review]
Updates good. One small nit. The example release note in the release checklist doc points to google storage.

**Comment by bacongobbler** [comment]
Good catch. Fixed!

File: docs/release_checklist.md [hunk 1]
```diff
@@ -149,24 +124,24 @@ git push upstream $RELEASE_CANDIDATE_NAME
 
 CircleCI will automatically create a tagged release image and client binary to test with.
 
-For testers, the process to start testing after CircleCI finishes building the artifacts involves the following steps to grab the client from Google Cloud Storage:
+After CircleCI finishes building the artifacts, the use the following commands to fetch the client for testing:
```
Comment (hickeyma) [inline docs/release_checklist.md]: small nit here if you want to change it: :s/the use/use

**Review by hickeyma — APPROVED** [review]
LGTM, thanks.

File: docs/release_checklist.md (hunk 1 above)
Comment (bacongobbler) [reply docs/release_checklist.md]: fixed. Thanks again

**Pull request merged**